aiida_diff/parsers.py

```diff
@@ -42,29 +40,19 @@ def parse(self, **kwargs):
         """
         from aiida.orm import SinglefileData
 
-        # Check that the retrieved folder is there
-        try:
-            output_folder = self.retrieved
-        except exceptions.NotExistent:
-            return self.exit_codes.ERROR_NO_RETRIEVED_FOLDER
-
-        # Check the folder content is as expected
-        list_of_files = output_folder.list_object_names()
-        output_files = [self.node.get_option('output_filename')]
-        output_links = ['diff']
-        # Note: set(A) <= set(B) checks whether A is a subset
-        if set(output_files) <= set(list_of_files):
-            pass
-        else:
-            self.logger.error(
-                "Not all expected output files {} were found".format(
-                    output_files))
+        output_filename = self.node.get_option('output_filename')
 
-        # Use something like this to loop over multiple output files
-        for fname, link in zip(output_files, output_links):
+        # Check that folder content is as expected
+        files_retrieved = self.retrieved.list_object_names()
+        files_expected = [output_filename]
+        # Note: set(A) <= set(B) checks whether A is a subset of B
+        if not set(files_expected) <= set(files_retrieved):
+            self.logger.error("Found files '{}', expected to find '{}'".format(
+                files_retrieved, files_expected))
 
-            with output_folder.open(fname, 'rb') as handle:
-                node = SinglefileData(file=handle)
-            self.out(link, node)
+        # add output file
+        with self.retrieved.open(output_filename, 'rb') as handle:
+            output_node = SinglefileData(file=handle)
+        self.out('diff', output_node)
 
         return ExitCode(0)
```

Comment:
The return `ExitCode(0)` is not necessary, so best remove it I would say

Reply:
yeah - it was in there to avoid a linter warning ;-)
anyhow, I'll remove it and suppress the warning

Reply:
hm... in the end I kept it. 
I would agree with pylint that it's not consistent to return or not return.
Anyhow, it does not hurt.